src/network/networkd-network.c

```diff
@@ -1321,11 +1321,15 @@ int config_parse_required_for_online(
                 }
 
                 required = r;
-                s = LINK_OPERSTATE_DEGRADED;
+                range = LINK_OPERSTATE_RANGE_DEFAULT;
         }
 
         network->required_for_online = required;
-        network->required_operstate_for_online = s;
+
+        if (range.min != _LINK_OPERSTATE_INVALID)
+                network->required_operstate_for_online.min = range.min;
+        if (range.max != _LINK_OPERSTATE_INVALID)
+                network->required_operstate_for_online.max = range.max;
```

Comment:
This seems wrong. If I do:
```
RequiredForOnline=A:B
RequiredForOnline=:C
```
then the first assignment should be completely overwritten by the second one. With the two `if`s above, I'll get something like A:C at the end.

Reply:
Changed the logic to explicitly set the default value again if an operstate is empty.